BluRunspace/BluRunspace.cs

```diff
@@ -158,39 +158,50 @@ private string ProcessResult(Collection<PSObject> psObjects, string scriptBlock)
         {
             var output = "Execution of Script Block: " + scriptBlock.FormatForEventLog();
             var result = string.Empty;
-            switch (psObjects.Count)
+            try
             {
-                case 0:
-                    output += "Is completed successfully and returned null." + Environment.NewLine;
-                    break;
-
-                case 1:
-                    // Accept null as a valid osObject and BaseObject, so return null
-                    if (psObjects[0] == null || psObjects[0].BaseObject == null)
-                    {
+                switch (psObjects.Count)
+                {
+                    case 0:
                         output += "Is completed successfully and returned null." + Environment.NewLine;
-                    }
-                    else
-                    {
-                        output += "Is completed successfully and returned:" + Environment.NewLine;
-                        output += psObjects[0].BaseObject.ToString();
-                        result += psObjects[0].BaseObject;
-                    }
-                    break;
-
-                default:
-                    output += "Is completed successfully and returned:" + Environment.NewLine +
-                              Environment.NewLine;
-                    foreach (var pso in psObjects)
-                    {
-                        output += pso.BaseObject + Environment.NewLine;
-                        result += pso.BaseObject + Environment.NewLine;
-                    }
-                    break;
+                        break;
+
+                    case 1:
+                        // Accept null as a valid osObject and BaseObject, so return null
+                        if (psObjects[0] == null || psObjects[0].BaseObject == null)
+                        {
+                            output += "Is completed successfully and returned null." + Environment.NewLine;
+                        }
+                        else
+                        {
+                            output += "Is completed successfully and returned:" + Environment.NewLine;
+                            output += psObjects[0].BaseObject.ToString();
+                            result += psObjects[0].BaseObject;
+                        }
+                        break;
+
+                    default:
+                        output += "Is completed successfully and returned:" + Environment.NewLine +
+                                  Environment.NewLine;
+                        foreach (var pso in psObjects)
+                        {
+                            if (pso == null || pso.BaseObject == null)
```

Comment:
This is the change that prevents a null reference exception in case of empty result object in the list of objects